discovery-provider/src/tasks/index_tastemaker_notifications.py

```diff
@@ -0,0 +1,112 @@
+from datetime import datetime
+from typing import List
+
+from sqlalchemy import asc
+from src.models.notifications.notification import Notification
+from src.models.social.repost import Repost
+from src.models.social.save import Save
+from src.models.tracks.track import Track
+from src.utils.session_manager import SessionManager
+
+
+def create_tastemaker_group_id(user_id, repost_item_id):
+    return f"tastemaker:{user_id}:tastemaker_item_id:{repost_item_id}"
+
+
+def index_tastemaker_notifications(
+    db: SessionManager,
+    top_trending_tracks: List[Track],
+    tastemaker_notification_threshold=10,
+):
+    with db.scoped_session() as session:
+        tastemaker_notifications = []
+        for track in top_trending_tracks:
+            (
+                repost_tastemaker_notifications,
+                existing_group_ids,
+            ) = create_action_tastemaker_notifications(
+                tastemaker_notification_threshold,
+                session,
+                existing_group_ids=set(),
+                track=track,
+                action_type=Repost,
+            )
+            tastemaker_notifications.extend(repost_tastemaker_notifications)
+
+            (
+                save_tastemaker_notifications,
+                existing_group_ids,
+            ) = create_action_tastemaker_notifications(
+                tastemaker_notification_threshold,
+                session,
+                existing_group_ids=existing_group_ids,
```

Comment:
so in this case, we are using existing group_ids from the repost call above so we dont double make? i like the idea, i wonder if we can just run these separately and then do a uniqueness filter on group_id after the fact? it potentially makes the "create" function simpler, but all good either way :)

Reply:
i like this! let me give it a shot